conan/tools/microsoft/visual.py

```diff
@@ -65,9 +56,99 @@ def generate(self, scope="build"):
         create_env_script(conanfile, content, CONAN_VCVARS_FILE, scope)
 
 
+def _vcvars_path(version, vs_install_path):
+    # TODO: This comes from conans/client/tools/win.py vcvars_command()
+    vs_path = vs_install_path or vs_installation_path(version)
+    if not vs_path or not os.path.isdir(vs_path):
+        raise ConanException("VS non-existing installation: Visual Studio %s" % version)
+
+    if int(version) > 14:
+        vcpath = os.path.join(vs_path, "VC/Auxiliary/Build/vcvarsall.bat")
+    else:
+        vcpath = os.path.join(vs_path, "VC/vcvarsall.bat")
+    return vcpath
+
+
+def _vcvars_arch(conanfile):
+    """
+    Computes the vcvars command line architecture based on conanfile settings (host) and
+    settings_build.
+    """
+    settings_host = conanfile.settings
+    settings_build = conanfile.settings_build
+
+    arch_host = str(settings_host.arch)
+    arch_build = str(settings_build.arch)
+
+    arch = None
+    if arch_build == 'x86_64':
+        arch = {'x86': "amd64_x86",
+                'x86_64': 'amd64',
+                'armv7': 'amd64_arm',
+                'armv8': 'amd64_arm64'}.get(arch_host)
+    elif arch_build == 'x86':
+        arch = {'x86': 'x86',
+                'x86_64': 'x86_amd64',
+                'armv7': 'x86_arm',
+                'armv8': 'x86_arm64'}.get(arch_host)
+
+    if not arch:
+        raise ConanException('vcvars unsupported architectures %s-%s' % (arch_build, arch_host))
+
+    return arch
+
+
+def _vcvars_vers(conanfile, compiler, vs_version):
+    if int(vs_version) <= 14:
+        return None
+
+    assert compiler == "msvc"
+    # Code similar to CMakeToolchain toolset one
+    compiler_version = str(conanfile.settings.compiler.version)
+    # The equivalent of compiler 192 is toolset 14.2
+    vcvars_ver = "14.{}".format(compiler_version[-1])
+    return vcvars_ver
+
+
+def msvc_version_to_vs_ide_version(version):
```

Comment:
There are not publicly exported tools yet. Shouldn't be documented as part of the API

Reply:
Don't worry because I'll select ONLY the public ones for Conan docs. This docstring could help any new Conan developer.